Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-volumegroup-snapshot.md

```diff
@@ -0,0 +1,100 @@
+that Rook is running in. When Ceph CSI is deployed by Rook, the operator will automatically
+maintain a configmap whose contents will match this key. By default this is
+"rook-ceph".
+
+```console
+kubectl create -f deploy/examples/csi/cephfs/groupsnapshotclass.yaml
+```
+
+### VolumeGroupSnapshot
+
+In [VolumeGroupSnapshot](https://github.com/rook/rook/tree/master/deploy/examples/csi/cephfs/groupsnapshot.yaml),
+`volumeGroupSnapshotClassName` should be the name of the `VolumeGroupSnapshotClass`
+previously created. The labels inside `matchLabels` should be the present on the
+PVC which is already created by the CephFS CSI driver.
+
+```console
+kubectl create -f deploy/examples/csi/cephfs/groupsnapshot.yaml
+```
+
+### Verify CephFS GroupSnapshot Creation
+
+```console
+$ kubectl get volumegroupsnapshotclass
+NAME                              DRIVER                          DELETIONPOLICY   AGE
+csi-cephfsplugin-groupsnapclass   rook-ceph.cephfs.csi.ceph.com   Delete           21m
+```
+
+```console
+$ kubectl get volumegroupsnapshot
+NAME                       READYTOUSE   VOLUMEGROUPSNAPSHOTCLASS          VOLUMEGROUPSNAPSHOTCONTENT                              CREATIONTIME   AGE
+cephfs-pvc-groupsnapshot   true         csi-cephfsplugin-groupsnapclass   groupsnapcontent-77ef7c5b-93ae-4ec3-b880-fa70740fb681   2m27s          2m29s
+```
+
+The snapshot will be ready to restore to a new PVC when `READYTOUSE` field of the
+`volumegroupsnapshot` is set to true.
+
+### Restore the CephFS groupsnapshot to a new PVC
+
+In
+[pvc-group-restore](https://github.com/rook/rook/tree/master/deploy/examples/csi/cephfs/pvc-restoregroup.yaml),
+`dataSource` should be the name of the `VolumeGroupSnapshot` previously
+created. The `dataSource` kind should be the `VolumeSnapshot`.
+
+Create a new PVC from the snapshot
+
+```console
+kubectl create -f deploy/examples/csi/cephfs/pvc-restoregroup.yaml
+```
+
+### Verify CephFS Restore PVC Creation
+
+```console
+$ kubectl get pvc
+NAME                      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   VOLUMEATTRIBUTESCLASS   AGE
+cephfs-pvc                Bound    pvc-9ae60bf9-4931-4f9a-9de1-7f45f31fe4da   1Gi        RWO            rook-cephfs    <unset>                 26m
+cephfs-pvc-restoregroup   Bound    pvc-9ae60bf9-4931-4f9a-9de1-7f45f31fe4da   1Gi        RW0            rook-cephfs    <unset>                 30s
+```
+
+## CephFS snapshot resource Cleanup
+
+To clean your cluster of the resources created by this example, run the following:
+
+```console
+kubectl delete -f deploy/examples/csi/cephfs/pvc-restoregroup.yaml
+kubectl delete -f deploy/examples/csi/cephfs/groupsnapshot.yaml
+kubectl delete -f deploy/examples/csi/cephfs/groupsnapshotclass.yaml
+```
+
+## Limitations
+
+- There is a limit of 400 snapshots per cephFS filesystem.
+- The PVC cannot be deleted if it has snapshots. make sure all the snapshots on the PVC are deleted before you delete the PVC.
```

Comment:
we dont have this limitation any more, we can remove this from here and from snapshot documentation as well